go/tools/gopackagesdriver/packageregistry.go

```diff
@@ -18,39 +18,64 @@ import (
 	"fmt"
 	"os"
 	"strings"
+
+	"golang.org/x/tools/go/packages"
 )
 
 type PackageRegistry struct {
-	packagesByID map[string]*FlatPackage
-	stdlib       map[string]string
+	packagesByID map[string]*packages.Package
+	stdlib       map[string]*packages.Package
 	bazelVersion bazelVersion
 }
 
 func NewPackageRegistry(bazelVersion bazelVersion, pkgs ...*FlatPackage) *PackageRegistry {
 	pr := &PackageRegistry{
-		packagesByID: map[string]*FlatPackage{},
-		stdlib:       map[string]string{},
+		packagesByID: map[string]*packages.Package{},
+		stdlib:       map[string]*packages.Package{},
 		bazelVersion: bazelVersion,
 	}
 	pr.Add(pkgs...)
 	return pr
 }
 
 func (pr *PackageRegistry) Add(pkgs ...*FlatPackage) *PackageRegistry {
-	for _, pkg := range pkgs {
+	for _, flatPkg := range pkgs {
+		imports := make(map[string]*packages.Package)
+		for impKey, imp := range flatPkg.Imports {
+			imports[impKey] = &packages.Package{
+				ID: imp,
+			}
+		}
+
+		pkg := &packages.Package{
+			ID:              flatPkg.ID,
+			Name:            flatPkg.Name,
+			PkgPath:         flatPkg.PkgPath,
+			GoFiles:         flatPkg.GoFiles,
+			CompiledGoFiles: flatPkg.CompiledGoFiles,
+			OtherFiles:      flatPkg.OtherFiles,
+			ExportFile:      flatPkg.ExportFile,
+			Imports:         imports,
+		}
+
+		// TODO(ellie): wtf why does this fix things
```

Comment:
Do we know why now?

Reply:
My guess would be here: https://github.com/bazel-contrib/rules_go/blob/e8adba26a407fbea9354a6d88f6f97bf3ae1106c/go/tools/builders/stdliblist.go#L275 I haven’t tested this yet, but it would line up here — we build everything with `pure` (i.e. no cgo)

Reply:
Just checked - this was why this was actually broken.